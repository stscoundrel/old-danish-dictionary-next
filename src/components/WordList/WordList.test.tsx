import ReactDOM from 'react-dom/client'
import renderer from 'react-test-renderer'
import WordList from './index'

const entries = [
  {
    headword: 'Grønfugl',
    definitions: [
      'no. totanus. Moth. Smlgn. grittue.',
    ],
    slug: 'gronfugl',
  },
  {
    headword: 'Grønfugl 2',
    definitions: [
      'no. totanus. Moth. Smlgn. grittue.',
    ],
    slug: 'gronfugl',
  },
  {
    headword: 'Grønfugl 3',
    definitions: [
      'no. totanus. Moth. Smlgn. grittue.',
    ],
    slug: 'gronfugl',
  },
  {
    headword: 'Grønfugl 4',
    definitions: [
      'no. totanus. Moth. Smlgn. grittue.',
    ],
    slug: 'gronfugl',
  },
]

describe('WordList component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<WordList words={entries} />)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<WordList words={entries} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Renders correct amount of words', () => {
    const tree = renderer.create(<WordList words={entries} />)
    const { root } = tree

    expect(root.findAllByType('ul').length).toEqual(1)
    expect(root.findAllByType('li').length).toEqual(4)
  })
})
