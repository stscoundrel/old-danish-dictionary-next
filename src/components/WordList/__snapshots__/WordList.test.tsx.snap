// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordList component Matches snapshot 1`] = `
<ul
  className="list"
>
  <li>
    <a
      className="link"
      href="/word/gronfugl"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      grønfugl
    </a>
  </li>
  <li>
    <a
      className="link"
      href="/word/gronfugl"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      grønfugl 2
    </a>
  </li>
  <li>
    <a
      className="link"
      href="/word/gronfugl"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      grønfugl 3
    </a>
  </li>
  <li>
    <a
      className="link"
      href="/word/gronfugl"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      grønfugl 4
    </a>
  </li>
</ul>
`;
